### PR TITLE
Cleaning up xref, finalizing things

### PR DESCRIPTION
**Pull request by sidvishnoi**

### CODE REVIEW

File: src/core/link-to-dfn.js [hunk 1]
```diff
@@ -243,7 +246,7 @@ function findExplicitExternalLinks() {
       const closest = el.closest("[data-cite]");
       return !closest || closest.dataset.cite !== "";
     })
-    .concat([...document.querySelectorAll("dfn.externalDFN")]);
+    .concat([...externalDFNs]);
```
Comment (marcoscaceres) [inline src/core/link-to-dfn.js]:
```suggestion
    .concat(...externalDFNs);
```

**Comment by sidvishnoi** [comment]
This looks ready now. Rest of xref changes are based on `core/inlines`. Will add some tests here though

File: src/core/xref.js [hunk 2]
```diff
@@ -384,23 +422,23 @@ function disambiguate(fetchedData, context, term) {
...
+    if (valid && contextFor && entry.for) {
       valid = entry.for.includes(contextFor);
     }
     return valid;
   });
 
   if (!data.length) {
     // no match found
-    return { error: "NO_MATCH", term };
+    throw { error: "NO_MATCH", term };
```
Comment (sidvishnoi) [inline src/core/xref.js]: TODO: create a XrefError class

File: src/core/xref.js [hunk 3]
```diff
@@ -384,23 +422,23 @@ function disambiguate(fetchedData, context, term) {
...
   }
 
   if (data.length === 1) {
     return data[0]; // unambiguous
   }
 
   // ambiguous
-  return { error: "AMBIGUOUS", term, specs: data.map(e => e.spec) };
+  throw { error: "AMBIGUOUS", term, specs: data.map(e => e.spec) };
```
Comment (marcoscaceres) [inline src/core/xref.js]: Chatted with @sidvishnoi about making this into real `Error`  by making a new error class.

File: src/core/xref.js [hunk 4]
```diff
@@ -302,42 +341,47 @@ function isNormative(elem) {
...
     for (const entry of entries) {
-      const result = disambiguate(results[term], entry, term);
-      const { elem } = entry;
-      if (result.error) {
-        collectErrors(term, elem, result, errorCollectors);
+      let result;
+      try {
+        result = disambiguate(results[term], entry, term);
+      } catch (error) {
+        collectErrors(entry.elem, error, errorCollectors);
```
Comment (sidvishnoi) [inline src/core/xref.js]: I think we should add elem in `error.data` also. WDYT?

**Pull request merged**